dask_kubernetes/tests/test_async.py

```diff
@@ -786,3 +786,29 @@ async def test_start_with_workers(pod_spec, ns, auth):
         async with Client(cluster, asynchronous=True) as client:
             while len(cluster.scheduler_info["workers"]) != 2:
                 await asyncio.sleep(0.1)
+
+
+@pytest.mark.asyncio
+async def test_adapt_delete(cluster, ns):
+    cluster.adapt(maximum=2, minimum=2)
+
+    start = time()
+    while len(cluster.scheduler_info["workers"]) != 2:
+        await asyncio.sleep(0.1)
+        assert time() < start + 20
+
+    core_api = cluster.core_api
+    pods_list = await core_api.list_namespaced_pod(ns)
+    worker_pods = [
+        x.metadata.name
+        for x in pods_list.items
+        if x.metadata.name.startswith(cluster.name)
+    ]
+    assert len(worker_pods) == 2
+    # delete one worker pod
+    await core_api.delete_namespaced_pod(name=worker_pods[0], namespace=ns)
+    # test whether adapt will bring it back
+    start = time()
+    while len(cluster.scheduler_info["workers"]) != 2:
+        await asyncio.sleep(0.1)
+        assert time() < start + 20
```

Comment:
It might be nice to explicitly assert that there are two worker pods after you've waited

Reply:
all fixed